### ISSUE TITLE
resolved_library_path_ never gets set

### ISSUE THREAD

**Issue by cottsay** [ros/pluginlib#15]
I have little knowledge of this, but I'm pretty sure @dirk-thomas (or someone else) will know for sure if this is a bug or if I'm using pluginlib wrong.

In my implementation I cannot let pluginlib manage instance. The problem is that when I try to unload the library (as the documentation says I need to do manually), pluginlib finds that `resolved_library_path_` is `UNRESOLVED` and refuses to call the internal unload routine. The plugin is therefore never unloaded.

From what I can tell, the line that set resolved_library_path_ was removed in https://github.com/ros/pluginlib/commit/e16b58d658a750ea85418c2a552e4d1e54ffe0ad and doesn't appear to be set anywhere else.

As a simple test:
- create an unmanaged instance of a plugin
- delete the reference to said instance
- check if library is loaded with `isClassLoaded` (this should still be true)
- attempt to unload the nodelet using `unloadLibraryForClass`

This throws pluginlib::LibraryUnloadException because `resolved_library_path_ == "UNRESOLVED"`

Was there a change that isn't in the documentation or is this a bug?

Thanks in advance for any guidance anyone can provide.

**Comment by mirzashah** [comment]
hey @cottsay, I'm the current maintainer of pluginlib. I'm looking into this now. If you could provide a simple test program that would be great. The unit tests for pluginlib and the underlying class_loader package should handle this situation.

**Comment by mirzashah** [comment]
You're right...it isn't being set any more. Not sure why that was removed and how things work without it. I'll investigate further.

**Comment by cottsay** [comment]
Thanks for getting back to me so quickly, @mirzashah . I created a quick test package available at https://github.com/cottsay/pluginlib_test .

**Comment by mirzashah** [comment]
Fixed in 560ccbe9188f22f76c8bca40d5703b104569073b. Simple fix to set that value. I don't think anybody noticed as the library is unloaded via another mechanism on ClassLoader destruction (destruction of the member class_loader::MultiLibraryClassLoader object takes care of this). 

I've added a unit test for this case and realized I need to add in a few more unit tests. Let me know if this version works before I do a release.
